# Analysis of Identification of lincRNA as predictive biomarkers in breast cancer

## ABSTRACT

Breast cancer is the most common malignant neoplasia in women ([WHO](https://www.who.int/cancer/prevention/diagnosis-screening/breast-cancer/en/)). 
For that reason, there are many clinical strategies diagnosis and treatment-decision tools based principally in detection of proteins by 
immunhistochemical techniques or by PCR analysis (to detect mRNA) (Patani, 2013). However, less than %50 of the patients could respond to 
treatment in the clinical outcome, generating the necessity of better oncological management. In that sense, the development of better techniques is the new focus of traslational medicine (Alcaraz, 2017) and many of them are based in molecular biology techniques. One of this techniques is RNA Next Generation Sequencing
(RNA-Seq, KÃ¼lahoglu, 2014), which provide us the information of all transcriptomic context of a given sample, like a breast tumor sample, 
and allowed us to make clinical decision (Berger, 2018).

In here, we analized 10 samples from 10 patients diagnosed with breast cancer which where candidates to receive neoadjuvant treatment, 
classified into 2 groups according to clinical outcome data: No response patients (8) and Response to neoadjuvant treatment patients (2). 
Samples were took before pharmacological administration and the RNA and DNA was extracted for RNA-Seq and Whole Exome Sequencing 
implementation, in order to determine a tumoral molecular signature that can be used in clinical routine to predict response to neoadjuvant 
treatment, composed by genomic information (Copy number variants or CNV's) and transcriptomic data (lincRNA differentially expressed). Although 
we couldn't obtain yet CNV's information, we found a set of lincRNA that are differentially expressed in No responsive group and which
could be use to determine, by their own, response to neoadjuvant treatment.

## DISCUSSION

The oncological management of breast cancer patients depends on the progression of disease and on the 
